azext_edge/edge/providers/check/akri.py

```diff
@@ -0,0 +1,598 @@
+                            f"Configuration name: [cyan]{configuration_name}[/cyan]",
+                            (0, 0, 0, 16),
+                        ),
+                    )
+
+                shared = spec.get("shared", False)
+                check_manager.add_display(
+                    target_name=target_instances,
+                    namespace=namespace,
+                    display=Padding(
+                        f"Shared: [cyan]{str(shared)}[/cyan]",
+                        (0, 0, 0, 16),
+                    ),
+                )
+
+                if detail_level == ResourceOutputDetailLevel.verbose.value:
+                    broker_properties = spec.get("brokerProperties", {})
+                    if broker_properties:
+                        process_dict_resource(
+                            check_manager=check_manager,
+                            target_name=target_instances,
+                            resource=broker_properties,
+                            namespace=namespace,
+                            padding=16,
+                            prop_name="Broker properties",
+                        )
+
+                    # nodes
+                    nodes = spec.get("nodes", [])
+                    for node in nodes:
+                        check_manager.add_display(
+                            target_name=target_instances,
+                            namespace=namespace,
+                            display=Padding(
+                                f"Node: [cyan]{node}[/cyan]",
+                                (0, 0, 0, 16),
+                            ),
+                        )
+
+                    # deviceUsage
+                    device_usage = spec.get("deviceUsage", {})
+                    if device_usage:
+                        process_dict_resource(
+                            check_manager=check_manager,
+                            target_name=target_instances,
+                            resource=device_usage,
+                            namespace=namespace,
+                            padding=16,
+                            prop_name="Device usage",
+                        )
+
+    return check_manager.as_dict(as_list)
+
+
+def _validate_one_of_conditions(
+        conditions: List[tuple],
+        check_manager: CheckManager,
+        eval_value: dict,
+        namespace: str,
+        target_name: str,
+) -> None:
+    if len(conditions) == 1:
+        return
+
+    non_empty_conditions_count = len([condition for condition in conditions if condition[1]])
+
+    eval_status = CheckTaskStatus.success.value
+    conditions_names = ", ".join([f"'{condition[0]}'" for condition in conditions])
+    if non_empty_conditions_count == 0:
+        check_manager.add_display(
+            target_name=target_name,
+            namespace=namespace,
+            display=Padding(
+                f"One of {conditions_names} should be specified",
+                (0, 0, 0, 16),
```

Comment:
For these nested methods, it might be a little more modular if we pass in a starting `padding` value from where it's called. I see that each padding in here starts around 16-20, but if we add or remove any padding in the parent method, we may need to adjust padding in multiple places here.

One strategy I like is keeping track of the current padding level at the top level of the checks, and passing the current "level" of padding for the evaluation methods to start at:
```
padding = 4
check_top_level(padding=padding)
detail_padding = padding + 4
for item in detail_items:
    check_detail_level(item=item, padding=detail_padding)
    check_more_detail(item=item, padding=detail_padding+4)
...
etc
```